src/Security/Authorization/Policy/src/AuthorizationMiddleware.cs

```diff
@@ -35,8 +35,13 @@ public async Task Invoke(HttpContext context)
 
             var endpoint = context.GetEndpoint();
 
-            // Flag to indicate to other systems, e.g. MVC, that authorization middleware was run for this request
-            context.Items[AuthorizationMiddlewareInvokedKey] = AuthorizationMiddlewareInvokedValue;
+            if (endpoint != null)
```

Comment:
A comment about why the flag is only set if an endpoint is present

Reply:
What happens if no endpoint is matched? i.e. a 404

I'm guessing that UseEndpoint will no-op, and when it no-ops it will skip checking this flag, and the response will 404. Add an integration test for this secnario

Reply:
Added a couple. It ends up no-oping since the check is predicated on the discovered endpoint having CORS or auth metadata.